### PR TITLE
chore: update renovate config

### PR DESCRIPTION
**Pull request by lengau**
- [x] Have you followed the guidelines for contributing?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `tox`?

-----

### CODE REVIEW

File: .github/renovate.json5 [hunk 1]
```diff
@@ -10,7 +10,7 @@
       // Automerge patches, pin changes and digest changes.
       // Also groups these changes together.
       groupName: "bugfixes",
-      excludePackagePrefixes: ["dev", "lint", "types"],
+      excludePackagePrefixes: ["lint", "types"],
```
Comment (lengau) [inline .github/renovate.json5:13]: Excluding dev here made `requirements.txt` and `requirements-dev.txt` get out of sync sometimes

File: .github/workflows/check-renovate.yaml
Comment (lengau) [inline .github/workflows/check-renovate.yaml:1]: This additional workflow allows us to check renovate changes in CI.

File: .github/renovate.json5 [hunk 2]
```diff
@@ -91,12 +89,12 @@
...
-  automergeSchedule: ["every weekend"],
   schedule: ["every weekend"],
   prConcurrentLimit: 2, // No more than 2 open PRs at a time.
   branchConcurrentLimit: 20, // No more than 20 open branches at a time.
   prCreation: "not-pending", // Wait until status checks have completed before raising the PR
   prNotPendingHours: 4, // ...unless the status checks have been running for 4+ hours.
   prHourlyLimit: 1, // No more than 1 PR per hour.
-  stabilityDays: 2 // Wait 2 days from release before updating.
+  stabilityDays: 2, // Wait 2 days from release before updating.
+  automergeStrategy: "squash" // Squash & rebase when auto-merging.
```
Comment (lengau) [inline .github/renovate.json5:99]: Aligns with out standard merge strategy.

**Review by syu-w — APPROVED** [review]

**Comment by syu-w** [comment]
Who can use the ssh access?

**Comment by lengau** [comment]
@syu-w I'm not sure who exactly, but it's anyone who's marked as a contributor and who starts the run. So you or I can use the SSH access if we start a run, but a random person can't.

**Pull request merged**